.github/workflows/soroban-rpc.yml

```
@@ -61,21 +61,16 @@ jobs:
         with:
           go-version: 1.20.1
 
-      - if: matrix.os == 'windows-latest'
-        name: Find the minimal supported rust version on windows
-        run: |
-          preflight_rust_version=$(sed -n 's/.*rust-version = "\(.*\)"/\1/p' cmd/soroban-rpc/lib/preflight/Cargo.toml)
-          echo "PREFLIGHT_RUST_VERSION=$preflight_rust_version" >> $GITHUB_ENV
-      - name: Update rust toolchain as needed
-        run: |
-          rustup update $PREFLIGHT_RUST_VERSION
       - run: rustup target add ${{ matrix.rust_target }}
+
+      # On windows, make sure we have the same compiler (linker) used by rust.
+      # This is important since the symbols names won't match otherwise.
       - if: matrix.os == 'windows-latest'
-        name: Configure preflight library to use explicit rust version
+        name: Install the same mingw gcc compiler used by rust
         run: |
-          cd cmd/soroban-rpc/lib/preflight
-          rustup override set $PREFLIGHT_RUST_VERSION-${{ matrix.rust_target }}
-          cd ../../../..
+          C:\\msys64\\usr\\bin\\pacman.exe -S mingw-w64-x86_64-gcc --noconfirm
+          echo "CC=C:\\msys64\\mingw64\\bin\\gcc.exe" >> $GITHUB_ENV
+          echo "C:\\msys64\\mingw64\\bin" >> $GITHUB_PATH
```

would be cleaner to just use the envvar you set :)

```suggestion
          echo "C:\\msys64\\mingw64\\bin" >> $GITHUB_PATH
          $GITHUB_PATH\\pacman.exe -S mingw-w64-x86_64-gcc --noconfirm
          echo "CC=$GITHUB_PATH\\gcc.exe" >> $GITHUB_ENV
```

but just a nitpick

---

(also I'm almost positive Windows allows `/` over `\\` but again, just a nit)